lib/cli-engine/cli-engine.js

```diff
@@ -680,11 +684,13 @@ class CLIEngine {
 
         results.forEach(result => {
             const filteredMessages = result.messages.filter(isErrorMessage);
+            const filteredSuppressedMessages = result.suppressedMessages.filter(isErrorMessage);
 
-            if (filteredMessages.length > 0) {
+            if (filteredMessages.length > 0 || filteredSuppressedMessages.length > 0) {
```

Comment:
This would change the behavior of this method and consequently the behavior of the `--quiet` CLI flag, which returns only lint results with active (nonsuppressed) errors.

Reply:
So it should be
```js
if (filteredMessages.length > 0) {
    filtered.push({
        ...result,
        messages: filteredMessages,
        suppressedMessages: filteredSuppressedMessages,
        errorCount: filteredMessages.length,
        warningCount: 0,
        fixableErrorCount: result.fixableErrorCount,
        fixableWarningCount: 0
    });
}
```
right? If there is not an active error, `ESLint.getErrorResults()` will return an empty list. If there are active errors, it will filter out all warnings and return both of the active and suppressed errors.

Reply:
Yes, exactly that.